src/prof65/main.c

```diff
@@ -0,0 +1,198 @@
+/*                                                                           */
+/* 1. The origin of this software must not be misrepresented; you must not   */
+/*    claim that you wrote the original software. If you use this software   */
+/*    in a product, an acknowledgment in the product documentation would be  */
+/*    appreciated but is not required.                                       */
+/* 2. Altered source versions must be plainly marked as such, and must not   */
+/*    be misrepresented as being the original software.                      */
+/* 3. This notice may not be removed or altered from any source              */
+/*    distribution.                                                          */
+/*                                                                           */
+/*****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static unsigned samples[64 * 1024];
+
+struct label_t {
+    char name[32];
+    unsigned short addr;
+    unsigned count;
+};
+
+#define MAX_LABELS 1024
+static struct label_t labels[MAX_LABELS];
+static unsigned numlabels;
+
+static int byaddr(const void *ap, const void *bp)
+{
+    const struct label_t *a = (struct label_t *) ap;
+    const struct label_t *b = (struct label_t *) bp;
+
+    if (a->addr < b->addr) {
+        return -1;
+    }
+    if (a->addr > b->addr) {
+        return 1;
+    }
+    return 0;
+}
+
+static int bycount(const void *ap, const void *bp)
+{
+    const struct label_t *a = (struct label_t *) ap;
+    const struct label_t *b = (struct label_t *) bp;
+
+    if (a->count > b->count) {
+        return -1;
+    }
+    if (a->count < b->count) {
+        return 1;
+    }
+    return 0;
+}
+
+static struct label_t *findlabel(const unsigned short addr)
+{
+    unsigned i;
+    for (i = 1; i < numlabels; i++) {
+        if (labels[i - 1].addr <= addr && labels[i].addr > addr) {
+            return &labels[i - 1];
+        } else if (i == numlabels - 1 && labels[i].addr <= addr) {
+            return &labels[i];
+        }
+    }
+
+    return NULL;
+}
+
+static void usage(const char *argv0)
+{
+    printf ("Usage: %s [args] profile label\n\n"
+            "-v\tVerbose\n", argv0);
+    exit (0);
+}
+
+int main(int argc, char **argv)
+{
+    unsigned char verbose = 0;
+    char buf[1024];
+    const char *profile = NULL, *label = NULL;
+    int opt;
+    unsigned i, total;
+
+    while ((opt = getopt (argc, argv, "hv")) != -1) {
+        switch (opt) {
+            case 'h':
+            default:
+                usage (argv[0]);
+            break;
+            case 'v':
+                verbose = 1;
+            break;
+        }
+    }
+
+    if (optind + 2 > argc) {
+        usage (argv[0]);
+    }
+
+    profile = argv[optind];
+    label = argv[optind + 1];
+
+    /* samples */
+    FILE *f = fopen (profile, "rb");
```

Comment:
Defining a variable here is C99 style and won't compile with a C89/C90 compiler.
As we insist on C89'/90 style comments, should we insist on C89/C90 style var definitions?

Reply:
> All declarations in a block must be at the beginning of that block.

yes, we do :)